File: fizzbuzz/fizz_buzz_ruby/spec/fizzbuzz_spec.rb
require_relative '../lib/fizzbuzz'


describe 'fizzbuzz' do
  it 'does it return fizz if multiple of three' do
    expect(fizzbuzz(3)).to eq 'fizz'
  end
end

describe 'fizzbuzz' do
  it 'does it return fizz if multiple of five' do
    expect(fizzbuzz(5)).to eq 'buzz'
  end
end

describe 'fizzbuzz' do
  it 'if passed 6, return fizz as 6 is multiple of 3' do
    expect(fizzbuzz(6)).to eq 'fizz'
  end
end

describe 'fizzbuzz' do
  it 'if passed 10, return buzz as 10 is multiple of 5' do
    expect(fizzbuzz(10)).to eq 'buzz'
  end
end

describe 'fizzbuzz' do
  it 'if passed number not multiple of 3 or 5' do
    expect(fizzbuzz(4)).to eq 4
  end
end

describe 'fizzbuzz' do
  it 'if passed number not multiple of 3 or 5' do
    expect(fizzbuzz(4)).to eq 4
  end
end

describe 'fizzbuzz' do
  it 'if passed multiple of 3 and 5 then return fizzbuzz' do
    expect(fizzbuzz(15)).to eq 'fizzbuzz'
  end
end

File: fizzbuzz/fizz_buzz_js/spec/fizzBuzzSpec.js
describe('FizzBuzz', function() {
  var fizzBuzz;
  beforeEach(function() {
    fizzBuzz = new FizzBuzz();
  });

  describe('knowns when a number is', function() {
    it('is divisible by three', function() {
      expect(fizzBuzz._isDivisibleBy(3, 3)).toBe(true);
    });
    it('is divisible by five', function() {
      expect(fizzBuzz._isDivisibleBy(5, 5)).toBe(true);
    });
    it('is divisible by three and five', function() {
      expect(fizzBuzz._isDivisibleByFifteen(15)).toBe(true);
    });
  });

  describe('knowns when a number is not', function() {
    it('is not divisible by three', function() {
      expect(fizzBuzz._isDivisibleBy(2, 3)).toBe(false);
    });
    it('is not divisible by five', function() {
      expect(fizzBuzz._isDivisibleBy(2, 5)).toBe(false);
    });
    it('is not divisible by three and five', function() {
      expect(fizzBuzz._isDivisibleByFifteen(2)).toBe(false);
    });
  });

  describe('when playing says fizz buzz', function() {
    it('says FizzBuzz when div of 15', function () {
      expect(fizzBuzz.says(15)).toEqual('FizzBuzz')
    });
    it('says Fizz instead of 3', function() {
      expect(fizzBuzz.says(3)).toEqual('Fizz')
    });
    it('says Buzz instead of 5', function () {
      expect(fizzBuzz.says(5)).toEqual('Buzz')
    });
    it('returns the number if not 3/5/15', function () {
      expect(fizzBuzz.says(2)).toEqual(2)
    });
  });
});